config.hh.cmake

```diff
@@ -13,15 +13,13 @@
 # define @LIBRARY_NAME@_MINOR_VERSION @PROJECT_VERSION_MINOR_CONFIG@
 # define @LIBRARY_NAME@_PATCH_VERSION @PROJECT_VERSION_PATCH_CONFIG@
 
-// Since i <= 0 is always true, we'll get a compiler warning if major, minor or patch versions are equal to 0.
-// To avoid throwing an error on otherwise valid code, we explicitly perform a signed <= comparison
-#define @LIBRARY_NAME@_VERSION_AT_LEAST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION>major || ((int)@LIBRARY_NAME@_MAJOR_VERSION>=(int)major && \
-                                                             (@LIBRARY_NAME@_MINOR_VERSION>minor || ((int)@LIBRARY_NAME@_MINOR_VERSION>=(int)minor && \
-                                                                                                     (int)@LIBRARY_NAME@_PATCH_VERSION>=(int)patch))))
+#define @LIBRARY_NAME@_VERSION_AT_LEAST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION>major || (@LIBRARY_NAME@_MAJOR_VERSION>=major && \
+                                                             (@LIBRARY_NAME@_MINOR_VERSION>minor || (@LIBRARY_NAME@_MINOR_VERSION>=minor && \
+                                                                                                     @LIBRARY_NAME@_PATCH_VERSION>=patch))))
```

Comment:
Not sure why this isn't
```

#define @LIBRARY_NAME@_VERSION_AT_LEAST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION>major || (@LIBRARY_NAME@_MAJOR_VERSION==major && \
                                                             (@LIBRARY_NAME@_MINOR_VERSION>minor || (@LIBRARY_NAME@_MINOR_VERSION==minor && \
                                                                                                     @LIBRARY_NAME@_PATCH_VERSION>=patch))))
```
which will trigger one warning instead of 3.

For the *patch* check, I think the following equivalent code will not trigger a warning
`@LIBRARY_NAME@_PATCH_VERSION==patch || @LIBRARY_NAME@_PATCH_VERSION>patch`